src/CompileLessCommand.ts

```diff
@@ -45,6 +52,14 @@ export default class CompileLessCommand {
     }
   }
 
+  public setPreprocessor(preprocessor: Configuration.Preprocessor | undefined) {
+    if (preprocessor) {
+      this.document = preprocessor(this.document);
+      this.hasUsedPreprocessor = true;
+    }
+    return this;
+  }
+
```

Comment:
I like this fluent-api! Great idea. Can it please accept an array of preprocessors and rather than immediately processing the document, can it store the preprocessors in a class variable.

The correct time to apply them should be during execute. Unless you have good reasons to do otherwise, the input to preprocessor function should be a string, and it should return (I think) a promise of a string, so that it allows the consumer to do whatever they want.